client/src/document.js

```diff
@@ -133,47 +133,74 @@ function SidebarLeaf({ depth, title, content }) {
   );
 }
 
-function RenderHTMLElementDocument({ document }) {
-  let sections = [];
-
-  sections.push(
-    <Prose key="short_description" section={document.prose.short_description} />
-  );
-  sections.push(
-    <InteractiveExample key="interactive_example" document={document} />
-  );
-  sections.push(<Prose key="overview" section={document.prose.overview} />);
-  sections.push(<Attributes key="attributes" document={document} />);
-  sections.push(
-    <ProseWithHeading key="usage_notes" section={document.prose.usage_notes} />
-  );
-  sections.push(
-    <ProseWithHeading
-      key="accessibility_concerns"
-      section={document.prose.accessibility_concerns}
-    />
-  );
-  sections.push(<Examples key="examples" document={document} />);
-  sections.push(
-    <BrowserCompatibility key="browser_compatibility" document={document} />
-  );
-  sections.push(
-    <ProseWithHeading key="see_also" section={document.prose.see_also} />
-  );
+function RenderDocumentBody({ document }) {
+  const sections = [];
+  /**
+   * The reason we can't return a filtered map of 'document.body' is because
+   * some of the parts of 'document.body' is an array.
+   * E.g. document.body.additional_prose == [
+   *     {type: 'prose', value: STUFF},
+   *     {type: 'prose', value: OTHER_STUFF},
+   * ]
+   */
+  document.body.forEach((section, i) => {
+    if (section.type === "prose") {
```

Comment:
I'm no JS expert, so I may be off on what's idiomatic here, but I was thinking this long `if … else if … else` thing might be more readable as an object where the keys are the types and the values are functions that handle each type. Then you do one lookup for the handler, or go to a fallback if undefined. Though that might easier when the additional prose sections are more consumable

Reply:
That sounds reasonable. Leave as is 👍